### ISSUE TITLE
People: Face keeps getting linked to person even after removal

### ISSUE THREAD

**Issue by kevincox** [photoprism/photoprism#3654]
#### 1. What is not working as documented?

I have a false-positive association where a face is being added to the wrong cluster. (That is fine, it happens) However when I remove that face the next index adds the label back.

#### 2. How can we reproduce it?

Unfortunately I can not share the image publicly. Maybe I can share with an individual member or two. This is a small group of images. It is an east-Asian face with 3 images. A 4th image keeps getting added back but is a false-positive.

#### 3. What behavior do you expect?

Once an incorrect tag is removed it should never be automatically added back.

#### 4. What could be the cause of your problem?

Photoprism thinks that the face should match and is not learning from the negative feedback.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

```
time="2023-08-19T17:45:55Z" level=info msg="faces: face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 has ambiguous subject at dist 0.059342, Ø 0.810000 from 6 samples, collision Ø 0.268770"
time="2023-08-19T17:45:55Z" level=info msg="faces: successful conflict resolution for 'REDACTED', face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 had collisions with other persons"
time="2023-08-19T17:45:55Z" level=info msg="faces: face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 has ambiguous subject at dist 0.736918, Ø 0.810000 from 6 samples, collision Ø 0.049342"
time="2023-08-19T17:45:55Z" level=info msg="faces: successful conflict resolution for 'REDACTED', face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 had collisions with other persons"
time="2023-08-19T17:45:55Z" level=info msg="faces: face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 has ambiguous subject at dist 0.278770, Ø 0.810000 from 6 samples, collision Ø 0.726918"
time="2023-08-19T17:45:55Z" level=info msg="faces: successful conflict resolution for 'REDACTED', face SXDGQHMEEUGGD3TRY5VVS4KZIUETLSG3 had collisions with other persons"
time="2023-08-19T17:47:40Z" level=info msg="faces: resolved 3 / 3 ambiguous subjects [8m13.637879317s]"
time="2023-08-19T17:47:46Z" level=info msg="faces: found 2 new clusters"
time="2023-08-19T17:47:46Z" level=info msg="faces: skipped cluster 7CSOPQNYHYB6UTUJW4LMV2FTYJRIR4E2, embedding not distinct enough"
time="2023-08-19T17:47:46Z" level=info msg="faces: added 1 new faces [6.268577844s]"
time="2023-08-19T17:47:46Z" level=info msg="faces: found 12 unmatched markers"
time="2023-08-19T17:47:53Z" level=info msg="faces: resolved 42 conflicts"
time="2023-08-19T17:47:53Z" level=warning msg="faces: marker mrz3h2w1v6j75das face VLJOQKRSA7JR33VWMHP6X7ECF2BV5B3W has ambiguous subjects jrz5bx73br3ttqbc <> jrz2v3p2z60ivdl4, subject source manual"
time="2023-08-19T17:47:53Z" level=info msg="faces: resolved 13 conflicts"
time="2023-08-19T17:47:53Z" level=warning msg="faces: marker mrz3h2w31dfju4zr face VLJOQKRSA7JR33VWMHP6X7ECF2BV5B3W has ambiguous subjects jrz5eni14xqwtrn7 <> jrz2v3p2z60ivdl4, subject source manual"
time="2023-08-19T17:47:59Z" level=info msg="faces: resolved 10 conflicts"
time="2023-08-19T17:47:59Z" level=warning msg="faces: marker mrz42yl947xkp37r face VLJOQKRSA7JR33VWMHP6X7ECF2BV5B3W has ambiguous subjects jrzah1g31o2lp03o <> jrz2v3p2z60ivdl4, subject source manual"
time="2023-08-19T17:48:09Z" level=info msg="faces: updated 337 markers, recognized 331 faces, 9 unknown [22.579842311s]"
time="2023-08-19T17:48:09Z" level=info msg="upload: deleted empty folder /var/lib/photoprism/users/urz2unh2ciwq9e6z/upload/sessbu1vle55af34ji"
time="2023-08-19T17:48:09Z" level=info msg="upload: imported 21 files"
time="2023-08-19T17:48:10Z" level=info msg="upload has been processed"
```

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: PhotoPrism® CE Build 230615-90a18f6e7

(b) Database Type & Version: SQlite 3.42.0 2023-05-16 12:36:15 831d0fb2836b71c9bc51067c49fee4b8f18047814f2ff22d817d25195cf350b0

(c) Operating System Types & Versions: Linux

(d) Browser Types & Versions: N/A

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? N/A

#### 7. On what kind of device is PhotoPrism installed?

This is especially important if you are reporting a performance, import, or indexing issue. You can skip this if you're reporting a problem you found in our public demo, or if it's a completely unrelated issue, such as incorrect page layout.

(a) Device / Processor Type: AMD Ryzen 5 4600G

(b) Physical Memory & Swap Space in GB: 64GiB, 0 swap

(c) Storage Type: SSD

(d) Anything else that might be helpful to know?

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

NGINX, but this occurs without any HTTP traffic.

**Comment by lastzero** [comment]
This is very strange indeed, because manually changed faces should not be automatically assigned to a person. Some kind of database dump for the faces, makers and subjects tables would definitely help. I don't know how else we can figure out what is wrong in this case.

**Comment by kevincox** [comment]
Can you provide info about what relations should be dumped? I can identify the photo that is being added back and both face groups. 

I have tried giving the face a dummy name to the unknown face and that seems to be working as a workaround.

**Comment by lastzero** [comment]
@kevincox I'm sorry for not getting back to you sooner! The easiest way to create an index SQL dump is to [run the backup command](https://docs.photoprism.app/user-guide/backups/#backup-command) in a terminal:

```
docker compose exec photoprism photoprism backup -i -f
```

This should create a database dump in your `storage/backup` folder, which you can [share with us privately via email or with a download URL](https://www.photoprism.app/contact#file-samples). Alternatively, you can use SQLite directly as described in our advanced backup guide:

https://docs.photoprism.app/getting-started/advanced/backups/#sqlite-backups

Please Note:

- We only need the subjects, faces and markers tables for debugging. So if you want, you can remove the rest of the data from the generated SQL dumps (they are in text format, not binary) or modify the dump command to include only these tables.
- I could not find any information about the PhotoPrism version you have installed. If you are not using our latest stable version, I recommend you try it first before we start debugging your issue. Thank you very much!

**Comment by kevincox** [comment]
I have sent a copy of my database to samples@photoprism.app. Note that this is a recent dump after I have worked around this bug by creating a new person with the face. The temporary person that I have created is `subj_uid = 'jrzng8g3n2ahwav0'`.

The subject in question is `select * from subjects where subj_uid = 'jrz5bbg130ganvo2';`

I have performed some redaction on the files. However I would appreciate if these files could be shared with the minimum number of people and deleted as soon as possible.

1. `subj_slug`, `subj_name` in the `subjects` table have been replaced with a random value other than for the subject in question.
2. `marker_name` in the `markers` table has been replaced with a random value where it was non-empty other than the subject in question.
3. All tables but `subjects`, `faces` and `markers` have been dropped.

Let me know if additional information would help debugging.

**Comment by lastzero** [comment]
Thank you! There is a small chance that this fix will also solve your problem:

- #3414

Since you're using 230615, I'd recommend [updating](https://docs.photoprism.app/release-notes/#july-19-2023) in any case to see if that helps. We'll look into it as soon as possible, but need to finish the next stable release first (which is still a lot of work) and will then be traveling for a few days.

**Comment by Melsana** [comment]
I think I'm running into the same bug.  I have a girl and a boy who's faces have somehow been linked and when I change one it changes the other.

I name the girl, then go to the boy and click the eject button because it shows the girls name.  So I then put in the boys name then I go back to the previous picture of the girl and it now shows the boys name not the girls name.  I can even go and eject all the pictures of the boy from the girls cluster, then go find OTHER pictures of the boy and name them, and it renames the girl to the boy again.

**Comment by lastzero** [comment]
You are welcome to test it again with these improvements:
- https://github.com/photoprism/photoprism/issues/5167

**Comment by kevincox** [comment]
I am no longer using Photoprism, but maybe the others who have commented can try to reproduce.

However the linked issue doesn't really seem relevant. Because the bug isn't that face recognition is merging two people. That will always happen at some rate. The real bug here is that after separating them they are merged back together. That should **never** happen even if the face recognition says that every face is the same person.

But anyways, I will be unsubscribing from this issue.

**Comment by lastzero** [comment]
You are right that the title of the linked issue does not explicitly mention such improvements. However, they are described in the "Manual Cluster Merging & Retained Markers" section of the issue description or the package's `README.md` file:
- https://github.com/photoprism/photoprism/blob/develop/internal/ai/face/README.md

Since you are no longer using our software and will not be able to test the improvements we made, I will go ahead and close this issue.
